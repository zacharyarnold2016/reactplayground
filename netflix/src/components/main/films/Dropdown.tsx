import React, { useCallback, useState } from "react";
import EditMovie from "../../filmInteraction/edit/EditMovie";
import DeleteMovie from "../../filmInteraction/delete/DeleteMovie";

const Dropdown = () => {
  const [edit, setEdit] = useState(false);
  const [deleted, setDeleted] = useState(false);

  const renderEdit = useCallback(() => {
    setEdit(!edit);
  }, [edit]);

  const renderDelete = useCallback(() => {
    setDeleted(!deleted);
  }, [deleted]);

  return (
    <div className="dropdown">
      <h2 className="dropdownOption" onClick={renderEdit}>
        Edit
        {edit ? <EditMovie /> : null}
      </h2>
      <h2 className="dropdownOption" onClick={renderDelete}>
        Delete
        {deleted ? <DeleteMovie /> : null}
      </h2>
    </div>
  );
};

export default Dropdown;
