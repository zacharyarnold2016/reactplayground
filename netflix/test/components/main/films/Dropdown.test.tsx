// @ts-nocheck

import { expect } from "@jest/globals";
import renderer from "react-test-renderer";
import Dropdown from "../../../../src/components/main/films/Dropdown";
import * as redux from "react-redux";

jest.mock("react-redux");
const spyFunction = jest.fn(() => 1);
redux.useDispatch.mockReturnValue(spyFunction);

describe("Dropdown", () => {
  afterAll(() => {
    jest.unmock("react-redux");
  });
  test("onClick Edit", () => {
    const component = renderer.create(<Dropdown id={1} />);
    const tree = component.toJSON();
    tree.children[0].props.onClick();
    expect(spyFunction).toBeCalled();
  });
  test("onClick Delete", () => {
    const component = renderer.create(<Dropdown id={1} />);
    const tree = component.toJSON();
    tree.children[1].props.onClick();
    expect(spyFunction).toBeCalled();
  });
  test("Snapshot Test", () => {
    const component = renderer.create(<Dropdown id={1} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
