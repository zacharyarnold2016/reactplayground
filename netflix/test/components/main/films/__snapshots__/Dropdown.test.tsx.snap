// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown Snapshot Test 1`] = `
<div
  className="dropdown"
>
  <h2
    className="dropdownOption"
    onClick={[Function]}
  >
    Edit
  </h2>
  <h2
    className="dropdownOption"
    onClick={[Function]}
  >
    Delete
  </h2>
</div>
`;
